src/audio/buffers/comp_buffer.c

```diff
@@ -212,8 +212,8 @@ static struct comp_buffer *buffer_alloc_struct(void *stream_addr, size_t size, u
 	audio_stream_set_underrun(&buffer->stream, !!(flags & SOF_BUF_UNDERRUN_PERMITTED));
 	audio_stream_set_overrun(&buffer->stream, !!(flags & SOF_BUF_OVERRUN_PERMITTED));
 
-	list_init(&buffer->source_list);
-	list_init(&buffer->sink_list);
+	comp_buffer_reset_source_list(buffer);
+	comp_buffer_reset_sink_list(buffer);
```

Comment:
each change in this commit is simple, but there are so many of them... Maybe this could be split in 1 commit per accessor?